src/util/irep_ids.def

```diff
@@ -842,6 +842,7 @@ IREP_ID_ONE(statement_list_instructions)
 IREP_ID_ONE(max)
 IREP_ID_ONE(min)
 IREP_ID_ONE(constant_interval)
+IREP_ID_ONE(ptr_diff)
```

Comment:
I realise it is quite an elegant solution to the problem but I would ideally like to avoid adding an `irept` type that is a restricted version of another type / echos some of the type information.  I haven't got that far through the PR, maybe there is something that is simply not possible without it but, could we just have a utility method for `is this subtract a pointer difference`?

Reply:
Yep - this and all the scaffolding to support it is gone.